File: Etudes/LOTS/projetLot/SCALA/src/informatique/Reseau.scala
package informatique
import general.Graphe
import scala.collection.mutable.ArrayBuffer

class Reseau extends Graphe{
  type V = Ordinateur;
  type U = Cable;

  arets = new ArrayBuffer[U]();
  sommets = new ArrayBuffer[V]();
  
  def addSommet(sommet: V): String = {
    return sommet.setGraphe(this);
  }

  def addAret(aret: U): String = {
    return aret.setGraphe(this);
  }

}

File: Etudes/LOTS/projetLot/CSharp/CSharp/Informatique/Reseau.cs
using System;using System.Text;using System.Collections;using System.Collections.Generic;using General;namespace Informatique{  public class Reseau : Graphe  {      public Reseau(String name) {
		this.name=name;
	}

	
	override public bool compatibleAret(Aret aret) {
		return (aret is Cable);
	}

	override public bool compatibleSommet(Sommet sommet) {
		return (sommet is Ordinateur);
	}
  }}  // end of namespace Informatique